multimap/src/main/java/org/infinispan/multimap/impl/EmbeddedMultimapCache.java

```diff
@@ -157,39 +150,57 @@ public CompletableFuture<Boolean> containsEntry(K key, V value) {
 
    @Override
    public CompletableFuture<Long> size() {
-      try {
-         // block on explicit tx
-         return isExplicitTxContext() ?
-              completedFuture(sizeInternal()) :
-              supplyAsync(this::sizeInternal);
-      } catch (SystemException e) {
-         throw CompletableFutures.asCompletionException(e);
-      }
-   }
-
-   private boolean isExplicitTxContext() throws SystemException {
-      TransactionManager transactionManager = cache.getAdvancedCache().getTransactionManager();
-      return transactionManager != null && transactionManager.getTransaction() != null;
-   }
+      return sizeInternal().toCompletableFuture();
+   }
+
+   private CompletionStage<Void> removeInternal(Predicate<? super V> p) {
```

Comment:
This requires the predicate to be marshallable, but that should be fine. Especially how this improves the operation to not pull all data.

Reply:
Good point. We could even check if it is marshallable even and then do it based on that, but probably isn't worth. I can just change it back to more similar to what we had before.

Reply:
Actually, I forgot! `computeIfPresent` already states it should be marshallable, so we are fine 😌